src/js/features/search/hooks.ts

```diff
@@ -1,3 +1,22 @@
-import { useAppSelector } from '@/hooks';
+import { useEffect } from 'react';
+import { useAppDispatch, useAppSelector } from '@/hooks';
+import { makeGetIndividualData } from '@/features/search/makeGetIndividualData.thunk';
+import { RequestStatus } from '@/types/requests';
 
 export const useSearchQuery = () => useAppSelector((state) => state.query);
+
+export const useIndividualData = (id: string) => {
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    dispatch(makeGetIndividualData(id));
+  }, [dispatch, id]);
+
+  const individualData = useAppSelector((state) => state.query.individualDataCache[id]);
+  return individualData;
+};
+
+export const useIsFetchingIndividualData = (id: string) => {
```

Comment:
this is not an ideal way to factor this hook - i've refactored it a better way in my branch which I will open as a sub-PR.